### PR TITLE
DOCS-2517: Add backlinks from other APIs

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: src/viam/app/data_client.py [hunk 1]
```diff
@@ -114,8 +114,9 @@ async def main():
         if __name__ == '__main__':
             asyncio.run(main())
 
-    """
 
+    For more information, see `Data Client <https://docs.viam.com/appendix/apis/data-client/>`_.
```
Comment (stuqdog) [inline src/viam/app/data_client.py:118]: (minor) Above the comment is "For more information, see {FOO} Client _API_", but here we just say "see Data Client". For consistency, should this be "Data Client API"?

File: src/viam/app/ml_training_client.py [hunk 2]
```diff
@@ -54,6 +54,8 @@ async def main():
 
         if __name__ == '__main__':
             asyncio.run(main())
+
+    For more information, see `ML Training Client <https://docs.viam.com/appendix/apis/ml-training-client/>`_.
```
Comment (stuqdog) [inline src/viam/app/ml_training_client.py:58]: See above re: "ML Training Client" vs. "ML Training Client API".

**Review by stuqdog — APPROVED** [review]
A minor comment (if changes are made no need to rerequest my review), otherwise lgtm!

File: src/viam/app/ml_training_client.py (hunk 2 above)
Comment (npentrel) [reply src/viam/app/ml_training_client.py]: changed

File: src/viam/app/data_client.py (hunk 1 above)
Comment (npentrel) [reply src/viam/app/data_client.py]: changed

File: src/viam/app/ml_training_client.py [hunk 3]
```diff
@@ -88,11 +103,13 @@ async def submit_training_job(
...
         Returns:
             str: the id of the training job
+
+        For more information, see `ML Training Client API <https://docs.viam.com/appendix/apis/ml-training-client/>`_.
         """
 
         request = SubmitTrainingJobRequest(
             dataset_id=dataset_id,
-            organization_id=org_id,
+            org_id=org_id,
```
Comment (npentrel) [inline src/viam/app/ml_training_client.py:112]:
```suggestion
            organization_id =org_id,
```

File: src/viam/app/ml_training_client.py [hunk 4]
```diff
@@ -88,11 +103,13 @@ async def submit_training_job(
...
         Returns:
             str: the id of the training job
+
+        For more information, see `ML Training Client API <https://docs.viam.com/appendix/apis/ml-training-client/>`_.
         """
 
         request = SubmitTrainingJobRequest(
             dataset_id=dataset_id,
-            organization_id=org_id,
+            organization_id =org_id,
```
Comment (npentrel) [inline src/viam/app/ml_training_client.py:112]:
```suggestion
            organization_id=org_id,
```

File: src/viam/app/ml_training_client.py [hunk 5]
```diff
@@ -106,6 +123,16 @@ async def submit_custom_training_job(
     ) -> str:
         """Submit a custom training job.
 
+        ::
+
+            job_id = await ml_training_client.submit_custom_training_job(
+                org_id=organization_id,
```
Comment (npentrel) [inline src/viam/app/ml_training_client.py:129]:
```suggestion
                organization_id=organization_id,
```

**Review by purplenicole730 — APPROVED** [review]

**Review by purplenicole730 — COMMENTED** [review]
Do we not have backlinks for the provisioning client?

**Comment by npentrel** [comment]
We have no docs for the provisioning client in the main docs (yet)

**Pull request merged**